File: CokeLee777/src/com/leetcode/greedy/JumpGame.java
package com.leetcode.greedy;

/**
 * https://leetcode.com/problems/jump-game/
 */
public class JumpGame {

    public boolean mySourceCode(int[] nums) {
        int nowIdx = 0, lastIdx = nums.length-1;
        boolean[] canStay = new boolean[nums.length];
        //시작위치 방문처리
        canStay[0] = true;

        while(nowIdx <= lastIdx){
            if(!canStay[nowIdx]){
                nowIdx++;
                continue;
            }

            for(int i = nowIdx; i <= nowIdx + nums[nowIdx]; i++){
                if(i >= nums.length) break;
                canStay[i] = true;
            }

            nowIdx++;
        }

        return canStay[lastIdx];
    }

    public boolean solution(int[] A) {
        int max = 0;
        for(int i=0;i<A.length;i++){
            if(i>max) {return false;}
            max = Math.max(A[i]+i,max);
        }
        return true;
    }
}

File: dongdaang/greedy/Jump_Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if nums == [0]:
            return True
        elif nums[0] == 0:
            return False
        else:
            for i in range(1, len(nums)-1):
                if nums[i] == 0:
                    cnt = 0
                    for j in range(i):
                        if nums[j] > i - j:
                            break
                        else:
                            cnt += 1
                    if cnt == i:
                        return False
            return True
        
        
        #cnt 안쓰고 푸는 방법? -> 앞에서 하나만이라도 조건 만족하면 다음꺼 진행하는 코드 어떻게 짤까??